### PR TITLE
Segment new partner form [PD-1163]

### PR DESCRIPTION
**Pull request by tdphillips**
2015-03-23-164455_648x821_scrot

All tests pass.

### CODE REVIEW

**Comment by tdphillips** [comment]
@fezick I don't know if you wanted all of these labels to be changed, but I like how it turned out.

Everything in the first section would normally be prefixed with "Partner" while everything in the second would be prefixed with "Primary Contact."

**Comment by aspidites** [comment]
Being pedantic, but they are legitimate concerns:
- Technically, locations are optional, and you may have multiple. As such, I wonder if there should be a "Location" section
- Regardless, I think the partner information should be marked with a legend

I'm conflicted as to whether I should be impressed that you were able to add what needed to be added without rewriting the form, or sad that you weren't motivated to rewrite the form and do things "correctly". 

Nice work overall.

File: mypartners/forms.py [hunk 1]
```diff
@@ -273,6 +272,18 @@ def save(self, commit=True):
...
         return None
 
+    def get_field_sets(self):
+        """
+        NewPartnerForm is a combination Partner and Contact form. As
+        self.fields has already been turned into an OrderedDict in __init__,
+        we can easily segment our form into fieldsets.
+        """
+        sections = self.fields.keys()[:4], self.fields.keys()[4:]
+        field_sets = [
```
Comment (aspidites) [inline mypartners/forms.py]: TIL a form's fields can be accessed using subscript/`dict`/`__getitem__` syntax.

**Comment by aspidites** [comment]
I suspect that "Source" is only obvious to the individual who requested the feature. I, as a new user (who probably didn't pay attention to the live training session), would have no clue what source meant. Perhaps help text?

**Comment by tdphillips** [comment]
You can't have multiple locations on a contact that I can see. It's also entirely optional.

**Comment by tdphillips** [comment]
@fezick Suggestions?

**Comment by aspidites** [comment]
Spoke with @fezick  directly, who addressed all of my questions. This is good to go, and I've checked it locally.

**Comment by fezick** [comment]
Yup, sorry. I didn't see this until @aspidites brought it my attention.

**Pull request merged**